lib/proxy/settings/global.rb

```diff
@@ -17,7 +18,7 @@ class Global < ::OpenStruct
       :log_buffer => 2000,
       :log_buffer_errors => 1000,
       :ssl_disabled_ciphers => [],
-      :tls_disabled_versions => [],
+      :tls_disabled_versions => ['1.0'],
```

Comment:
Hmm now I see your point. Shall we fix the installer rather than this? My concern is older RHEL clients. We can hardcode this in the installer.

Reply:
I am keeping this, we do want to keep this setting, what if 1.2 is deprecated in the future? Then we would need to re-add this. This is consistent experience. If you have concerns about our installer setting an empty value, let's fix that.

I am going to add a warning by Smart Proxy if this array is empty.

Reply:
RHEL clients shouldn't connect to the proxy (over SSL). We've been disabling TLS 1.0 and older since 41293caaf08337e1e95379920e991b5dc4114830 (508a9e8b18d8811f65f0f9f4201c287bae55da2d also disabled TLS v1.1), so at least for 4 years. There is no reason to care about insecure protocols.

Reply:
Yes, SSL is disabled, no discussion. TLS on the other hand was disabled in the code for some time, then we have introduced the setting in d5f0f0043f49dd1f6b790. I would like to stick with this, it's better for the future to let our users to have opportunity to disable them as we add it later on down the road.

Reply:
But TLS v1.0 has been disabled for 4 years. You are introducing a setting to enable it again. This should not be an option because it's insecure. Since [2018-06-30](https://blog.pcisecuritystandards.org/are-you-ready-for-30-june-2018-sayin-goodbye-to-ssl-early-tls) it must be disabled to be PCI Data Security Standard compliant.

Reply:
Can you highlight a line in any of the patches from the PR where I introduce such setting?

Reply:
https://github.com/theforeman/smart-proxy/pull/623/files#diff-b8fbcbffabe3ae5c9935bb21e3e94852R125. The discussion we had in https://github.com/theforeman/smart-proxy/pull/623/files#r386982122.

Reply:
I think you are talking about semantics and I am talking about setting. If I get you right, you comment on the fact that webrick currently (after your patch I merged this morning) disables 1.0 and 1.1 in the code. Then the setting can do additional TLS versions.

Puma code does not hardcode anything, setting can disable 1.0 and 1.1.

Although I think this should NOT be hardcoded and it's wrong because I smell like there might be somebody with old RHEL who might not want to do this, I am giving up and rebasing the change to follow what you want.

Reply:
I am indeed not complaining about the setting in general, but rather about what it can achieve. Normally I agree with you that everything should be configurable. TLS 1.0 in particular should always be disabled. 1.1 should be safe as well given we don't want to support any clients from old RHEL. AFAIK the only tools that connects to Foreman Proxy over TLS are Foreman, Smart Proxy Dynflow Core and HTTPBoot. All of them should support TLS 1.2.